src/extensions/nuget/Microsoft.DotNet.UpgradeAssistant.Extensions.NuGet/NuGetTransitiveDependencyIdentifier.cs

```diff
@@ -137,25 +151,11 @@ public TargetGraphLookup(RestoreTargetGraph graph)
 
             public bool Contains(NuGetReference key) => TryFind(key, out _);
 
-            public IEnumerator<NuGetReference> GetEnumerator()
-            {
-                foreach (var item in _graph.Flattened)
-                {
-                    var library = item.Data.Match.Library;
-
-                    yield return new(library.Name, library.Version.ToString());
-                }
-            }
+            public IEnumerator<NuGetReference> GetEnumerator() => _graph.Flattened.Select(item => new NuGetReference(item.Data.Match.Library.Name, item.Data.Match.Library.Version.ToString())).GetEnumerator();
```

Comment:
nit: this seems more difficult to read than the foreach (and probably allocates a bit more if that matters)